lib/_http_server.js

```diff
@@ -516,14 +516,23 @@ function connectionListener(socket) {
     }
 
     if (req.headers.expect !== undefined &&
-        (req.httpVersionMajor == 1 && req.httpVersionMinor == 1) &&
-        continueExpression.test(req.headers['expect'])) {
-      res._expect_continue = true;
-      if (EventEmitter.listenerCount(self, 'checkContinue') > 0) {
-        self.emit('checkContinue', req, res);
+        (req.httpVersionMajor == 1 && req.httpVersionMinor == 1)) {
+      if (continueExpression.test(req.headers.expect)) {
+        res._expect_continue = true;
+
+        if (EventEmitter.listenerCount(self, 'checkContinue') > 0) {
```

Comment:
Also while we're in here, `EventEmitter.listenerCount()` is deprecated, so we should just use `self.listenerCount('checkContinue');` instead and similarly for the other event below.